instrumentation/grpc-1.6/library/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/TracingClientInterceptor.java

```diff
@@ -55,8 +56,15 @@
         throw e;
       }
     }
-
-    SocketAddress address = result.getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR);
+    SocketAddress address = null;
+    try {
+      URI uri = new URI(null, next.authority(), null, null, null);
+      address =
+          InetSocketAddress.createUnresolved(
+              uri.getHost(), uri.getPort() == -1 ? 80 : uri.getPort());
+    } catch (Throwable e) {
+      // do nothing
+    }
```

Comment:
I'm not sure about defaulting to port 80, it seems like missing port could also mean 443, so better to leave it -1, which will lead to no `net.peer.port` captured (which is better than an incorrect port)

```suggestion
    try {
      URI uri = new URI(null, next.authority(), null, null, null);
      request.setRemoteAddress(InetSocketAddress.createUnresolved(uri.getHost(), uri.getPort()));
    } catch (URISyntaxException ignored) {
    }
```

Reply:
`InetSocketAddress.createUnresolved` will check the port and throw an exception when port is -1. So maybe giving the default port with 80 or 443 is better.